### ISSUE TITLE
Link layout issues

### ISSUE THREAD

**Issue by kevinrenskers** [Lambdo-Labs/MDText#3]
This is how the text is supposed to be laid out:

<img width="509" alt="Screen Shot 2020-03-08 at 11 55 39" src="https://user-images.githubusercontent.com/229384/76161438-e287d300-6133-11ea-9914-935a0171273b.png">

But this is what happens to links:

<img width="509" alt="Screen Shot 2020-03-08 at 11 55 26" src="https://user-images.githubusercontent.com/229384/76161436-e1ef3c80-6133-11ea-8209-ea5173fe3fd1.png">

<img width="509" alt="Screen Shot 2020-03-08 at 11 55 58" src="https://user-images.githubusercontent.com/229384/76161441-e3206980-6133-11ea-9e14-e553e8ad9fdf.png">

**Comment by mralexhay** [comment]
Also seeing this issue

**Comment by andr-ec** [comment]
Yes this is currently a limitation with SwiftUI. As soon as we add a modifier to text to allow for a clickable link, it becomes a view and we can no longer compose the text. So in this case links become their own line. We could fix this by dipping into UIKit for the links. 

Unfortunately I don’t have a ton of time to look at this at the moment, but contributions are welcome!

**Comment by brett-eugenelabs** [comment]
I had a go at writing my own CommonMark Text widget in SwiftUI and also hit this problem, as Andre mentioned it is due to the way SwiftUI works, hopefully we get more support for attributed strings in SwiftUI 2.0 or even a wrap around stack which means we could mix Buttons and Text.

**Comment by mralexhay** [comment]
@acarrera94 that makes sense, thanks for the response 👍🏻

**Comment by andr-ec** [comment]
Looks like this might be possible in swiftUI. From [blog](https://fivestars.blog/swiftui/flexible-swiftui.html) and [code](https://github.com/zntfdr/FiveStarsCodeSamples/tree/main/Flexible-SwiftUI).

**Comment by AppChampion** [comment]
I tried both possible solutions but they didn't seem to work. Maybe this would be a solution: [SwiftUI Links in Text](https://gist.github.com/emma-k-alexandra/4de70bdf8d3a9754894ea061c802a3c6)
